cpp/src/copying/slice.cu

```diff
@@ -31,6 +32,44 @@
 
 namespace cudf {
 namespace detail {
+
+template <typename ColumnView>
+ColumnView slice(ColumnView const& input,
+                 size_type begin,
+                 size_type end,
+                 rmm::cuda_stream_view stream)
+{
+  CUDF_EXPECTS(begin >= 0, "Invalid beginning of range.");
+  CUDF_EXPECTS(end >= begin, "Invalid end of range.");
+  CUDF_EXPECTS(end <= input.size(), "Slice range out of bounds.");
+
+  std::vector<ColumnView> children{};
+  children.reserve(input.num_children());
+  for (size_type index = 0; index < input.num_children(); index++) {
+    children.emplace_back(input.child(index));
+  }
+
+  size_type nulls =
+    input.null_count() == 0 ? 0 : cudf::detail::null_count(input.null_mask(), begin, end, stream);
```

Comment:
Can you explain what is happening here? If `input.null_count != 0`, why are we computing the null count? It reads to me as if it's already known. If there is some default initialization value here then perhaps this will read cleaner like this, implying that we are always choosing a known null count:
```suggestion
    input.null_count() == UNKNOWN_NULL_COUNT ? cudf::detail::null_count(input.null_mask(), begin, end, stream) : input.null_count();
```

Reply:
This is the `slice` function which returns a segment view of the input in the range `[begin, end)`. The return null count (`nulls`) is counted nulls in that segment.

Reply:
Got it, yes. I just don't like the pattern `... == 0 ? 0 : ...`. Then my suggestion is
```suggestion
    input.null_count() ? cudf::detail::null_count(input.null_mask(), begin, end, stream) : 0;
```